src/slam_toolbox_common.cpp

```diff
@@ -535,16 +564,31 @@ void SlamToolbox::publishVisualizations()
       boost::mutex::scoped_lock lock(smapper_mutex_);
       closure_assistant_->publishGraph();
     }
+    
+    // Check if pose graph publishing was requested (e.g., after loop closure)
+    bool was_requested = publish_pose_graph_requested_.load();
+    
+    if (was_requested) {
```

Comment:
```suggestion 
    if (publish_pose_graph_requested_.load()) {
```

A little more concise